pylinac/core/profile.py

```diff
@@ -210,6 +236,57 @@ def fwxm_center(self, x: int=50, interpolate: bool=False) -> Tuple[NumberLike, N
             fwxm_center_idx = int(round(fwxm_center_idx))
             return fwxm_center_idx, self.values[fwxm_center_idx]
 
+    @argue.bounds(x=(0, 100))
+    @argue.options(norm=('max', 'max grounded', 'cax', 'cax grounded'), interpolate=(True, False))
+    def distance_to_dose(self, x: int=50, norm='max grounded', interpolate=True) -> Tuple[float, float]:
+        """Return the width at X-Max, where X is the percentage height.
+
+        Parameters
+        ----------
+        x : int
+            The dose level of the profile. E.g. x = 50 is 50% height, 100 is the profile peak.
+            i.e. FWHM.
+        norm : str
+            The type of normalisation to apply:
+                'max'          : normalises to the profile maximum with no grounding
+                'max grounded' : grounds each side to the profile minimum and normalises to the profile maximum
+                'cax'          : normalises to the profile centre value with no grounding
+                'cax grounded' : grounds each side to the profile minimum and normalises to the profile centre value
+            Default is 'max_grounded' as this is the intrinsic method of the scipy sgnal.find_peak function.
+        interpolate : bool
+            Whether to interpolate the profile array values to get subpixel precision.
+
+        Returns
+        -------
+        left index, right index
+            The left and right indices of the dose level.
+        """
+
+        # prevent array from being grounded by setting ends to zero
+        if norm in ['max', 'cax']:
+            ydata = np.insert(self.values, 0, 0)
+            ydata = np.append(ydata, 0)
+        else:
+            ydata = self.values
+
+        # adjust x to give the equivalent level if peak val was at CAX
+        if norm == 'cax':
+            ylen = len(ydata)
+            if ylen % 2 == 0:           # ylen is even and central detectors straddle CAX
+                cax = (ydata[int(ylen/2)] + ydata[int(ylen/2) - 1])/2.0
+            else:                       # ylen is odd and we have a central detector
+                cax = ydata[int((ylen - 1)/2)]
+            ymax = ydata.max()
+            x = x*cax/ymax
+
+        _, peak_props = find_peaks(ydata, fwxm_height=x/100, max_number=1)
+        left = peak_props['left_ips'][0] if interpolate else int(round(peak_props['left_ips'][0]))
+        right = peak_props['right_ips'][0] if interpolate else int(round(peak_props['right_ips'][0]))
+        if norm in ['max', 'cax']:
+            left = left - 1
```

Comment:
What is the -1 for?

Reply:
When the normalisation is "max" or "cax" I force find_peaks to use the full height of the peak from zero by adding a zero array value at the beginning and the end of the profile. This means that any indices returned by find_peaks needs to be decremented by 1 to correspond to the original profile.